airflow/models/dagrun.py

```diff
@@ -487,7 +487,9 @@ def task_instance_scheduling_decisions(self, session: Session = None) -> TISched
         schedulable_tis: List[TI] = []
         changed_tis = False
 
-        tis = list(self.get_task_instances(session=session, state=State.task_states + (State.SHUTDOWN,)))
+        tis = list(
+            self.get_task_instances(session=session, state=State.task_states + tuple(State.terminated_states))
+        )
```

Comment:
It's not clear to me why `State.SHUTDOWN` wasn't in task_states before, but it looks like Andew fixed that in #15285, so this can just become

```suggestion
        tis = list(self.get_task_instances(session=session, state=State.task_states))
```

Reply:
Thanks. This is great.